backend/gateway/mux/util.go

```diff
@@ -0,0 +1,38 @@
+package mux
+
+import (
+	"net/http"
+	"reflect"
+	"strings"
+)
+
+// All modern browsers send `text/html` as the first MIME type in the Accept header for navigation requests.
+// This is naive and doesn't do any actual content negotiation, just checks if the client accepts text/html which
+// is good enough for our use case of determining whether the request originates from the SPA's XHR network client or
+// browser navigation.
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation/List_of_default_Accept_values
```

Comment:
:slightly_smiling_face: I said "modern browsers", I think it's enough unless we want to record browser compatibility in a more structured way throughout the project. But none of the frontend is going to work on IE8 anyways, since React doesn't support IE8 without some additional polyfills (not even considering the myriad of other frontend libraries we use).

The stylesheet request shouldn't matter, since the browser won't be able to get the main page to then fetch the stylesheet without going through the redirect and authing first. Worst case in the extreme race condition where the user requests the main page successfully, then the token expires before the browser can followup with the request for the CSS, they would just get unstyled content (obviously broken page) and need a refresh to fix it. We could pass more info about the request here to check if the original request was for CSS, because otherwise there's no way to differentiate, but personally I don't think it's worth it. Actually, that race would effect non-Opera browsers too, and it would have to be handled by client code in order to remediate it since redirect or not, the user won't get any CSS from the request.